src/nspawn/nspawn.c

```diff
@@ -976,14 +976,12 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
 
                 case ARG_UUID:
-                        r = sd_id128_from_string(optarg, &arg_uuid);
+                        r = id128_from_string_not_null(optarg, &arg_uuid);
+                        if (r == -ENXIO)
+                                return log_error_errno(r, "Machine UUID may not be all zeroes.");
```

Comment:
Should we really log the `ENXIO` here?

Reply:
I think we should log the error. This will fail the invocation of `nspawn`:
```console
$ build/systemd-nspawn --uuid 00000000000000000000000000000000           
Machine UUID may not be all zeroes.
```

Reply:
Yeah, we should definitely log the error, I'm more concerned about the `ENXIO`, since the previous code used `SYNTHETIC_ERRNO(EINVAL)` - not sure where the `ENXIO` [ends up showing](https://github.com/systemd/systemd/pull/26198#discussion_r1086761701) (if at all).

Reply:
It's logged as `ERRNO=ENXIO` field in the journal. I don't think anybody would look at this. `parse_argv()` is called from `run()`, and is then translated by the main function wrapper to the return value of `1`. But yeah, ENXIO doesn't make much sense, so I changed the patch now to translate it to -EINVAL.